### DISCUSSION TITLE
How to create a macro for "upright(bold(<input>))" in math mode?

### DISCUSSION THREAD

**Discussion by floristfromnowhere** [typst/typst#432] [Q&A] [answered]
I've seen this [discussion titled *How to define custom shortcuts such as "aa" for "upright(bold(a))"?*](https://github.com/typst/typst/discussions/365) and I'm looking for a more general solution. How do I create a macro for `upright(bold(<input>))` in math mode?

I tried

```#let mathbf(input) = {$upright(bold(#input))$}```

but it doesn't seem to work. How can I achieve this properly?

**Comment by Enivex** [comment, marked as answer]
What dpesn't work exactly? It looks fine on my end.

**Reply by floristfromnowhere** [reply]
I want to create a shortcut command so I don't have to write `upright(bold(<input>))` every time and my attempt at defining such a shortcut didn't work.

**Reply by laurmaedje** [reply]
This
```
#let mathbf(input) = {$upright(bold(#input))$}
$mathbf(a)$
```
gives the desired output for me.

**Reply by floristfromnowhere** [reply]
It didn't work yesterday for some reason but it seems to be working fine now.
